regen-network/regen-web
# feat: APP-24 create post

## Description

https://regennetwork.atlassian.net/browse/APP-24?atlOrigin=eyJpIjoiZTY0MWE1MDRiOTlhNDJjMzlkNTE1ODUzNDUyMDBhYjUiLCJwIjoiaiJ9
closes: #2266
(will also fix https://regennetwork.atlassian.net/browse/APP-190?atlOrigin=eyJpIjoiM2Q5ZDlhOTE0Yjc0NDUwNThmMmI0NTA1NzFjNDMyODgiLCJwIjoiaiJ9)

needs: https://github.com/regen-network/regen-server/pull/469

This doesn't include the "Attest" signing part of the flow since it's pending https://github.com/regen-network/regen-server/pull/454. Follow-up task to address that: https://regennetwork.atlassian.net/browse/APP-183?atlOrigin=eyJpIjoiZmM0ZWU5Njk5ODcwNGY3NWFkMmUwNjNmNGM0ZDgxZjQiLCJwIjoiaiJ9



---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...


### How to test



1. Log in from https://deploy-preview-2381--regen-marketplace.netlify.app/
2. Go to your projects. Try to to create a post using the new dedicated button on your projects cards.
3. Add different types of files with different types of location and different privacy settings. NB: when selecting a file, it's uploaded directly so there might be some time before you see the file name in the "edit a file" popup. @erikalogie and I were discussing some ways to improve that but we haven't come to a conclusion yet, we'll need to implement some UX tweaks to that when she's back.
4. "Create post" button should be disabled for draft projects or projects without location

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...
